src/storage/v2/disk/label_property_index.cpp

```diff
@@ -0,0 +1,272 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+/// TODO: clear dependencies
+
+#include "storage/v2/disk/label_property_index.hpp"
+#include "storage/v2/id_types.hpp"
+#include "storage/v2/inmemory/indices_utils.hpp"
+#include "storage/v2/property_value.hpp"
+#include "utils/disk_utils.hpp"
+#include "utils/exceptions.hpp"
+#include "utils/file.hpp"
+#include "utils/skip_list.hpp"
+
+namespace memgraph::storage {
+
+namespace {
+
+bool IsVertexIndexedByLabelProperty(const Vertex &vertex, LabelId label, PropertyId property) {
+  return utils::Contains(vertex.labels, label) && vertex.properties.HasProperty(property);
+}
+
+[[nodiscard]] bool ClearTransactionEntriesWithRemovedIndexingLabel(
+    rocksdb::Transaction &disk_transaction,
+    const std::map<Gid, std::vector<std::pair<LabelId, PropertyId>>> &transaction_entries) {
+  for (const auto &[vertex_gid, index] : transaction_entries) {
+    for (const auto &[indexing_label, indexing_property] : index) {
+      if (auto status = disk_transaction.Delete(
```

Comment:
I think you can apply here similar changes as in label_index.cpp throughout the file